### PR TITLE
Issue 5413 - Use ConstructorBased Injection for ObjectMapper

### PR DESCRIPTION
**Pull request by elizabetht**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [X] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by elizabetht** [comment]
@cbornet Kindly review the PR and let me know your comments - I have fixed your comments from earlier review

File: modules/swagger-codegen/src/main/resources/JavaSpring/apiController.mustache [hunk 1]
```diff
@@ -32,6 +32,12 @@ import javax.validation.Valid;
 @Controller
 {{#operations}}
 public class {{classname}}Controller implements {{classname}} {
+    ObjectMapper objectMapper;
```
Comment (cbornet) [inline modules/swagger-codegen/src/main/resources/JavaSpring/apiController.mustache]: Can be `final`

Comment (elizabetht) [inline modules/swagger-codegen/src/main/resources/JavaSpring/apiController.mustache]: @cbornet Fixed it. Please review the changes and let me know!

**Comment by elizabetht** [comment]
@cbornet I have merged master with this branch and resolved conflicts. Could you let me know if this PR is good to be merged?

**Comment by cbornet** [comment]
One small issue I see is that if the controller doesn't have any exmaple, we inject the ObjectMapper and don't use it.
Other than that LGTM.

**Comment by elizabetht** [comment]
@cbornet Yes, I saw that too. Do you have suggestions on how to fix this? Because examples is within the context of an operation and to enable/disable the ObjectMapper mapper injection at the class level, I can't seem to find a tag that is associated with examples globally.

**Comment by wing328** [comment]
> One small issue I see is that if the controller doesn't have any exmaple, we inject the ObjectMapper and don't use it.

IMO this enhancement can be done later after this PR is merged into master.

**Pull request merged**